gravitee-plugin-policy/src/main/java/io/gravitee/plugin/policy/internal/PolicyPluginHandler.java

```diff
@@ -66,14 +66,18 @@ protected PolicyPlugin create(Plugin plugin, Class<?> pluginClass) {
 
     private void determineProxyPhases(Class policyClass, PolicyPluginImpl entity) {
         if (!entity.manifest().properties().containsKey("proxy") && !entity.manifest().properties().containsKey("message")) {
-            Set<String> proxyPhases = new HashSet<>();
-            if (methodFoundFor(REQUEST_ANNOTATIONS, policyClass)) {
-                proxyPhases.add("REQUEST");
-            }
-            if (methodFoundFor(RESPONSE_ANNOTATIONS, policyClass)) {
-                proxyPhases.add("RESPONSE");
+            try {
```

Comment:
Ideally, there is a way to detect that the gateway ( or rest api ???) is running (via Spring), and we don't have to run this for AM...

Reply:
Currently, there is no such info as far as I know.
The only thing we have is the "applcation" name in the Node but it a string define in each project so in the current state, it is hard to adapt the behaviour for APIM/AM project without introducing project specifuc constants.